dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/fileresource/JCloudsFileResourceContentStore.java

```diff
@@ -298,6 +287,23 @@ public URI getSignedGetContentUri( String key )
         return httpRequest.getEndpoint();
     }
 
+    @Override
+    public void copyContent( String key, OutputStream output )
+        throws IOException, NoSuchElementException
+    {
+        if ( !blobExists( key ) )
+        {
+            throw new NoSuchElementException( "key '" + key + "' not found." );
+        }
+
+        InputStream in = getBlob( key ).getPayload().openStream();
```

Comment:
We could use try-with-resources syntax here I think, to ensure the stream is closed if the copy operation fails.

try ( InputStream in = getBlob( key ).getPayload().openStream() ) { ..